Netcracker/KubeMarine
# Implement dry-run for install/upgrade process

### Description
This improvement is to enable dry-run for installation and upgrade procedure of Kubemarine using the existing --without-act option
* This dry-run implementation will perform all the possible checks and generate all the possible files in the dumps directory to simulate the actual install/upgrade process.
* It will not make changes to actual nodes but perform all the possible checks by executing read commands.
* User can simulate the actual procedure by running a dry run before executing the actual procedure.

### Solution
Improvement
* 

### How to apply
Code change

### Test Cases

**TestCase 1**
Steps:
1. Configure a valid cluster.yaml for the all-in-one cluster.
2. Run kubemarine install --without-act and check if the process succeeds and configuration files are created in the dumps directory.

**TestCase 2**
Steps:
1. Pass invalid plugin  yaml template inside kubemarine/plugins/yaml directory.
2. Run kubemarine install --without-act and check if the process fails and configuration files are created in the dumps directory.

**TestCase 3**
Steps:
1. Configure a valid procedure.yaml for the upgrade.
2. Run kubemarine upgrade procedure.yaml --without-act and check if the process succeeds and configuration files are created in the dumps directory.


### Checklist
